core/cmd/on_chain_config_cmd.go

```diff
@@ -0,0 +1,399 @@
+package cmd
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+
+	"github.com/river-build/river/core/config"
+	"github.com/river-build/river/core/contracts/river"
+	. "github.com/river-build/river/core/node/base"
+	"github.com/river-build/river/core/node/crypto"
+	"github.com/river-build/river/core/node/infra"
+	. "github.com/river-build/river/core/node/protocol"
+)
+
+func printOnChainConfig(cmd *cobra.Command, args []string) error {
+	ctx := cmd.Context()
+	cfg := cmdConfig
+
+	blockchain, err := crypto.NewBlockchain(
+		ctx,
+		&cfg.RiverChain,
+		nil,
+		infra.NewMetricsFactory(nil, "river", "cmdline"),
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	config, err := crypto.NewOnChainConfig(
+		ctx,
+		blockchain.Client,
+		cfg.RegistryContract.Address,
+		blockchain.InitialBlockNum,
+		blockchain.ChainMonitor,
+	)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Current block: %d\n", config.ActiveBlock())
+
+	yaml, err := yaml.Marshal(config.Get())
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Config:\n%s\n", string(yaml))
+	return nil
+}
+
+func getOnChainConfig(cmd *cobra.Command, args []string) error {
+	ctx := cmd.Context()
+	cfg := cmdConfig
+	key := args[0]
+	valueType := ""
+	if len(args) > 1 {
+		valueType = args[1]
+		if valueType != "uint" && valueType != "uints" && valueType != "int" && valueType != "string" {
+			return RiverError(Err_INVALID_ARGUMENT, "invalid value type", "type", valueType)
+		}
+	} else {
+		valueType = crypto.AllKnownOnChainSettingKeys()[key]
+	}
+
+	blockchain, err := crypto.NewBlockchain(
+		ctx,
+		&cfg.RiverChain,
+		nil,
+		infra.NewMetricsFactory(nil, "river", "cmdline"),
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	caller, err := river.NewRiverConfigV1Caller(cfg.RegistryContract.Address, blockchain.Client)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	opts := &bind.CallOpts{
+		Context: ctx,
+	}
+	settings, err := caller.GetConfiguration(opts, crypto.HashSettingName(key))
+	if err != nil {
+		return err
+	}
+
+	if len(settings) == 0 {
+		return RiverError(Err_INTERNAL, "returned seetings are empty")
+	}
+
+	for _, s := range settings {
+		fmt.Printf("block: %d\n", s.BlockNumber)
+		switch valueType {
+		case "":
+			fmt.Printf("%s\n", hex.EncodeToString(s.Value))
+		case "uint":
+			num, err := crypto.ABIDecodeUint64(s.Value)
```

Comment:
`crypto.ABIDecodeUint256` supports bigger numbers

Reply:
but there is a max size that can fit in the actual Config struct, I changed types to ABI type names to avoid confusion, they are now generated directly from the struct.